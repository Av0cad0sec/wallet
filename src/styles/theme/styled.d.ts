import 'styled-components'
import { ThemeType } from 'grommet'

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends ThemeType {}
}
