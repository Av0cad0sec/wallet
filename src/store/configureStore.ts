import { routerMiddleware } from 'connected-react-router'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { createInjectorsEnhancer } from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'

import { createReducer, history } from './reducers'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.message,
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })
  const { run: runSaga } = sagaMiddleware

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)]

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ]

  const store = configureStore({
    reducer: createReducer(),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production' || process.env.PUBLIC_URL.length > 0,
    enhancers,
    preloadedState: state,
  })

  return store
}
