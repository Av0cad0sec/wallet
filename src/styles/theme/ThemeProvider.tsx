import { Grommet, ThemeType } from 'grommet'
import * as React from 'react'

import { useThemeSlice } from './slice'

const defaultGrommetTheme: ThemeType = {
  global: {
    font: {
      family: 'Rubik',
    },
  },
}
export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice()

  // const theme = useSelector(selectTheme);
  return (
    // <OriginalThemeProvider theme={theme}>
    <Grommet full theme={defaultGrommetTheme}>
      {React.Children.only(props.children)}
    </Grommet>
    // </OriginalThemeProvider>
  )
}
