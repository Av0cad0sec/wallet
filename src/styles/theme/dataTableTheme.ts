import { IDataTableStyles, ITheme } from 'react-data-table-component'

export const dataTableTheme = {
  background: { default: 'false' },
  highlightOnHover: {
    default: '#88888833',
    text: 'false',
  },
  text: {
    primary: 'false',
    secondary: 'false',
    disabled: 'false',
  },
  sortFocus: {
    default: 'false',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>

export const dataTableCustomStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '18px',
    },
  },
  rows: {
    style: {
      fontSize: '16x',
    },
  },
}
