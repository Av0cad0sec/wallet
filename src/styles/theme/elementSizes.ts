export const sidebarSmallSizeLogo = 30
export const sidebarMediumSizeLogo = 40
export const mobileToolbarZIndex = 3
export const mobileToolbarButtonsZIndex = mobileToolbarZIndex + 1
export const buildPreviewBannerZIndex = mobileToolbarZIndex - 1

export const mobileFooterNavigationHeight = '4rem'
