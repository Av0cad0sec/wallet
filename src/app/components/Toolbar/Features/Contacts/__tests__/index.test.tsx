import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { Contacts } from '../'

const renderComponent = (store: any) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <Contacts />
      </ThemeProvider>
    </Provider>,
  )

describe('<Contacts  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  it('should match snapshot', () => {
    const storeWithContacts = configureAppStore({
      contacts: {
        oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4: {
          address: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4',
          name: 'My Contact',
        },
      },
    })
    const { container } = renderComponent(storeWithContacts)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render empty state', () => {
    renderComponent(store)
    expect(screen.getByText('toolbar.contacts.emptyList')).toBeInTheDocument()
    expect(screen.getByText('toolbar.contacts.add')).toBeInTheDocument()
    expect(screen.queryByTestId('account-choice')).not.toBeInTheDocument()
  })

  it('should show Add Contact form overlay', async () => {
    renderComponent(store)
    await userEvent.click(screen.getByRole('button', { name: 'toolbar.contacts.add' }))
    expect(screen.getByPlaceholderText('toolbar.contacts.name')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('toolbar.contacts.address')).toBeInTheDocument()
  })
})
