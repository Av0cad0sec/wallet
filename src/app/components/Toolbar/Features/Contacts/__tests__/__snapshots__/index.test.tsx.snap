// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contacts  /> should match snapshot 1`] = `
.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border: solid 1px #0092f6;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
  padding: 12px;
  border-radius: 5px;
  cursor: pointer;
}

.c2:hover {
  background-color: #0092f6;
  color: #FFFFFF;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 90%;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-right: 12px;
  overflow-y: auto;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c8 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c8:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c8:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c8:focus > circle,
.c8:focus > ellipse,
.c8:focus > line,
.c8:focus > path,
.c8:focus > polygon,
.c8:focus > polyline,
.c8:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c8:focus::-moz-focus-inner {
  border: 0;
}

.c8:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c8:focus:not(:focus-visible) > circle,.c8:focus:not(:focus-visible) > ellipse,
.c8:focus:not(:focus-visible) > line,.c8:focus:not(:focus-visible) > path,
.c8:focus:not(:focus-visible) > polygon,.c8:focus:not(:focus-visible) > polyline,
.c8:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c8:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0092f6;
  color: #F8F8F8;
  border-radius: 4px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c10:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c10:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c10:focus > circle,
.c10:focus > ellipse,
.c10:focus > line,
.c10:focus > path,
.c10:focus > polygon,
.c10:focus > polyline,
.c10:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c10:focus::-moz-focus-inner {
  border: 0;
}

.c10:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible) > circle,.c10:focus:not(:focus-visible) > ellipse,
.c10:focus:not(:focus-visible) > line,.c10:focus:not(:focus-visible) > path,
.c10:focus:not(:focus-visible) > polygon,.c10:focus:not(:focus-visible) > polyline,
.c10:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c10:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {
  .c2 {
    border: solid 1px #0092f6;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {

}

<div
  class="c0"
  style="min-height: 100dvh;"
>
  <div
    class="c1"
  >
    <div
      aria-checked="false"
      class="c2"
      data-testid="account-choice"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
          data-testid="account-name"
        >
          <span
            class="c5"
          >
            My Contact
          </span>
        </div>
        <div
          class="c4"
        >
          <span
            class="c5"
          >
            <span
              class="notranslate"
              translate="no"
            >
              oasis1qq2v...sv0d5eq4
            </span>
          </span>
        </div>
        <div
          class="c6"
        >
          <div
            class="c3"
          >
            <div
              class="c7"
            >
              <button
                class="c8"
                type="button"
              >
                Manage
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c9"
  >
    <button
      class="c10"
      type="button"
    >
      Add Contact
    </button>
  </div>
</div>
`;
